### PR TITLE
feat: tpm2 based disk encryption

### PR DESCRIPTION
**Pull request by frezbo**
Support disk encryption using tpm2 and pre-calculated signed PCR values.

Fixes: #7266

### CODE REVIEW

File: internal/app/init/main.go [hunk 1]
```diff
@@ -58,6 +59,11 @@ func run() (err error) {
 		log.Printf("failed to seed from the TPM: %s", err)
 	}
 
+	// extend PCR 11 with enter-initrd
+	if err = tpm2.PCRExtent(constants.UKIMeasuredPCR, []byte("enter-initrd")); err != nil {
```
Comment (smira) [inline internal/app/init/main.go:63]: should we use a Phase constant here?

**Review by smira — APPROVED** [review]
:cool: I want to do a separate PR to bump Go to 1.20.6, but it's all good with some small nits

**Comment by frezbo** [comment]
> cool I want to do a separate PR to bump Go to 1.20.6, but it's all good with some small nits

yup, go bump needs a separate PR, i just wanted to test in CI

File: internal/app/init/main.go (hunk 1 above)
Comment (frezbo) [reply internal/app/init/main.go]: fixed

**Pull request merged**